packages/excalidraw/fonts.ts

```diff
@@ -0,0 +1,420 @@
+
+import PacificoCyrlicExt from "https://fonts.gstatic.com/s/pacifico/v22/FwZY7-Qmy14u9lezJ-6K6MmBp0u-zK4.woff2";
+import PacificoVietnamese from "https://fonts.gstatic.com/s/pacifico/v22/FwZY7-Qmy14u9lezJ-6I6MmBp0u-zK4.woff2";
+import PacificoLatinExt from "https://fonts.gstatic.com/s/pacifico/v22/FwZY7-Qmy14u9lezJ-6J6MmBp0u-zK4.woff2";
+import PacificoLatin from "https://fonts.gstatic.com/s/pacifico/v22/FwZY7-Qmy14u9lezJ-6H6MmBp0u-.woff2";
+
+import PermanentMarker from "https://fonts.gstatic.com/s/permanentmarker/v16/Fh4uPib9Iyv2ucM6pGQMWimMp004La2Cf5b6jlg.woff2";
+
+/** For head & hhea metrics read the woff2 with https://fontdrop.info/  */
+interface FontMetrics {
+  /** head.unitsPerEm */
+  unitsPerEm: 1000 | 1024 | 2048;
+  /** hhea.ascender */
+  ascender: number;
+  /** hhea.descender */
+  descender: number;
+  /** harcoded unitless line-height, https://github.com/excalidraw/excalidraw/pull/6360#issuecomment-1477635971 */
+  lineHeight: number;
+}
+
+export const FONT_METRICS: Record<number, FontMetrics> = {
+  [FONT_FAMILY.Virgil]: {
+    unitsPerEm: 1000,
+    ascender: 886,
+    descender: -374,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.Excalifont]: {
+    unitsPerEm: 1000,
+    ascender: 886,
+    descender: -374,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.Helvetica]: {
+    unitsPerEm: 2048,
+    ascender: 1577,
+    descender: -471,
+    lineHeight: 1.15,
+  },
+  [FONT_FAMILY.TeXGyreHeros]: {
+    unitsPerEm: 1000,
+    ascender: 1148,
+    descender: -284,
+    lineHeight: 1.15,
+  },
+  [FONT_FAMILY.Cascadia]: {
+    unitsPerEm: 2048,
+    ascender: 1900,
+    descender: -480,
+    lineHeight: 1.2,
+  },
+  [FONT_FAMILY.ComicShanns]: {
+    unitsPerEm: 1000,
+    ascender: 750,
+    descender: -250,
+    lineHeight: 1.2,
+  },
+  [FONT_FAMILY.Assistant]: {
+    unitsPerEm: 1000,
+    ascender: 1021,
+    descender: -287,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.Nunito]: {
+    unitsPerEm: 1000,
+    ascender: 1011,
+    descender: -353,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.Bangers]: {
+    unitsPerEm: 1000,
+    ascender: 883,
+    descender: -181,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.PermanentMarker]: {
+    unitsPerEm: 1024,
+    ascender: 1136,
+    descender: -325,
+    lineHeight: 1.25,
+  },
+  [FONT_FAMILY.Pacifico]: {
+    unitsPerEm: 1000,
+    ascender: 1303,
+    descender: -453,
+    lineHeight: 1.75,
+  },
+};
+
+export class ExcalidrawFont {
+  public readonly uri: string;
+  public readonly url: URL;
+  public readonly fontFace: FontFace;
+
+  constructor(family: string, uri: string, descriptors?: FontFaceDescriptors) {
+    this.uri = uri;
+    
+    // base urls will be applied for relative `uri`'s only
+    // TODO: should be loaded from a CDN once we publish the package (and the env variables are correctly injected)
+    this.url = new URL(uri, window.EXCALIDRAW_ASSET_PATH ?? window.origin);
```

Comment:
Rather than falling back to `window.origin` and expecting the host apps to copy the assets, we might want to fall back to our public CDN. However, atm vite env variables do not seem to be injected at build time (needs to be fixed first) & neither is the package publishes, which ends up with unloaded fonts i.e. in the nextjs example. Also, we should consider a separate CDN for long-lived static assets, such as fonts.